src/backend/app/Jobs/TokenProcessor.php

```diff
@@ -113,7 +113,12 @@ private function retryTokenGeneration(): void {
     }
 
     private function saveToken(array $tokenData): void {
-        $token = Token::query()->make(['token' => $tokenData['token'], 'load' => $tokenData['load']]);
+        $token = Token::query()->make(['token' => $tokenData['token'], 'token_amount' => $tokenData['load']]);
+        $token->device_id = $this->transactionContainer->device->id;
+        if ($this->transactionContainer->meter !== null) {
+            $token->token_type = Token::TYPE_ENERGY;
+            $token->token_unit = Token::UNIT_KWH;
+        }
```

Comment:
From what I understand the way `saveToken` is used is that it takes the output of the various manufacturer's `chargeDevice`.

The interface is already quite odd... For example AngazaSHS returns

```
        return [
            'token' => $token,
            'load' => $energy,
        ];
```

But here `load` is actually "days".

Should also demand the new-style token here? I.e. require `$tokenData` to be passed like this

```php
[
        'token' => $token,
        'token_type' => ...,
        'token_unit' => ...,
        'token_amount' => ...,
];
```

This would require changing all `chargeDevice` functions of all Providers, but would be a great change.